sherlockml/clients/job.py

```diff
@@ -0,0 +1,303 @@
+    subrun_number = fields.Integer(data_key="subrunNumber", required=True)
+    state = EnumField(SubrunState, by_value=True, required=True)
+    started_at = fields.DateTime(data_key="startedAt", missing=None)
+    ended_at = fields.DateTime(data_key="endedAt", missing=None)
+
+    @post_load
+    def make_subrun_summary(self, data):
+        return SubrunSummary(**data)
+
+
+class RunSchema(Schema):
+    id = fields.UUID(data_key="runId", required=True)
+    run_number = fields.Integer(data_key="runNumber", required=True)
+    state = EnumField(RunState, by_value=True, required=True)
+    submitted_at = fields.DateTime(data_key="submittedAt", required=True)
+    started_at = fields.DateTime(data_key="startedAt", missing=None)
+    ended_at = fields.DateTime(data_key="endedAt", missing=None)
+    subruns = fields.Nested(SubrunSummarySchema, many=True, required=True)
+
+    @post_load
+    def make_run(self, data):
+        return Run(**data)
+
+
+class ListRunsResponseSchema(Schema):
+    pagination = fields.Nested(PaginationSchema, required=True)
+    runs = fields.Nested(RunSummarySchema, many=True, required=True)
+
+    @post_load
+    def make_list_runs_response_schema(self, data):
+        return ListRunsResponse(**data)
+
+
+class EnvironmentStepExecutionSchema(Schema):
+    environment_id = fields.UUID(data_key="environmentId", required=True)
+    environment_step_id = fields.UUID(
+        data_key="environmentStepId", required=True
+    )
+    environment_name = fields.String(data_key="environmentName", required=True)
+    command = fields.String(required=True)
+    state = EnumField(RunState, by_value=True, required=True)
+    state = EnumField(
+        EnvironmentStepExecutionState, by_value=True, required=True
+    )
+    started_at = fields.DateTime(data_key="startedAt", missing=None)
+    ended_at = fields.DateTime(data_key="endedAt", missing=None)
+
+    @post_load
+    def make_environment_step_execution_schema(self, data):
+        return EnvironmentStepExecution(**data)
+
+
+class SubrunSchema(Schema):
+    id = fields.UUID(data_key="subrunId", required=True)
+    subrun_number = fields.Integer(data_key="subrunNumber", required=True)
+    state = EnumField(SubrunState, by_value=True, required=True)
+    started_at = fields.DateTime(data_key="startedAt", missing=None)
+    ended_at = fields.DateTime(data_key="endedAt", missing=None)
+    environment_step_executions = fields.Nested(
+        EnvironmentStepExecutionSchema,
+        data_key="environmentExecutionState",
+        many=True,
+        required=True,
+    )
+
+    @post_load
+    def make_subrun(self, data):
+        return Subrun(**data)
+
+
+class JobClient(BaseClient):
+
+    SERVICE_NAME = "steve"
+
+    def list(self, project_id):
+        endpoint = "/project/{}/job".format(project_id)
+        return self._get(endpoint, JobSummarySchema(many=True))
+
+    def create_run(self, project_id, job_id, parameter_value_sets):
```

Comment:
Should there be a docstring for creating a run? In particular, the structure of `parameter_value_sets` is not obvious.

Reply:
Not sure if it's conventional in this, but should we consider a default `parameter_value_sets=None` (obviously defaulting to `[{}]`)?